handler/windows: build /subsystem:windows by default. #19

KrzaQ opened the pull request:
Additionally add a separate /subsystem:console build for console programs and debugging purposes.

Finally, create a test Qt program to exemplify that those changes do work.

This change is a fix for #19.

vlussenburg commented on examples/windows/qt-window-app/CMakeLists.txt:
```diff
@@ -0,0 +1,34 @@
+    add_subdirectory(crashpad)
+endif ()
+
+find_package(Qt5 REQUIRED COMPONENTS Widgets)
```
Does this execute correctly on the GitHub pipeline? I'm not seeing the Qt executable (not sure if that is how it works).

Also, should the archive script be updated to archive the demo apps too if they're built?

KrzaQ replied:
I don't think we want examples in the archive, at least not in the one we currently have (which is used to disseminate pre-built libraries).

As for executing correctly in our CI pipeline: no, it does not, because we don't have Qt 5 available in there. But `examples/CMakeLists.txt` has only builds it if it's available now.

vlussenburg replied:
Agreed. Probably we just want to individually upload those example executables if they were built. Or an example apps archive or something. I think it’s useful to have for us/support/customers to be able to easily run one of these on specific target platforms

vlussenburg replied:
Could we apt-get the qt5 stuff?

KrzaQ replied:
We can install `qt5-default` or `qtbase5-dev qtchooser qt5-qmake qtbase5-dev-tools` (as per https://askubuntu.com/a/1335187)

KrzaQ committed:
handler/windows: build /subsystem:windows by default. Fixes #19

Additionally add a separate /subsystem:console build for console
programs and debugging purposes.

KrzaQ committed:
examples/windows: add a Qt widget example.

konraddysput commented on examples/windows/qt-window-app/backtrace.cpp:
```diff
@@ -0,0 +1,81 @@
+
+    database = crashpad::CrashReportDatabase::Initialize(db);
+
+    std::cerr << __LINE__ << '\n';
```
why do we need this?

KrzaQ replied:
We don't, but we're using it in every other example, so it's consistent. I'm open to removing it.

konraddysput commented on examples/windows/qt-window-app/backtrace.hpp:
```diff
@@ -0,0 +1,9 @@
+
+namespace backtrace
+{
+bool initialize_crashpad(std::string_view url);
```
should we add tabs here?

KrzaQ replied:
No, namespaces are not indented in any major C++ coding styles including Crashpad's.

konraddysput commented on examples/windows/qt-window-app/mainwindow.cpp:
```diff
@@ -0,0 +1,95 @@
+
+void MainWindow::selectHandlerPath()
+{
+    auto f = QFileDialog::getOpenFileName(this, "Select handler executable",
```
what is f?

KrzaQ replied:
The file name. It's type is `QString` but it's not pertinent here.
